persist-cli/src/main/java/io/ballerina/persist/utils/SqlScriptGenerationUtils.java

```diff
@@ -0,0 +1,420 @@
+import io.ballerina.persist.objects.Relation;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Sql script generator.
+ *
+ * @since 0.1.0
+ */
+public class SqlScriptGenerationUtils {
+
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String TAB = "\t";
+    private static final String EMPTY = "";
+    private static final String PRIMARY_KEY_START_SCRIPT = NEW_LINE + TAB + "PRIMARY KEY(";
+    private static final String UNIQUE_KEY_START_SCRIPT = NEW_LINE + TAB + "UNIQUE KEY(";
+    private static final String UNIQUE = " UNIQUE";
+    private static final String ON_DELETE_SYNTAX = " ON DELETE";
+    private static final String ON_UPDATE_SYNTAX = " ON UPDATE";
+    private static final String RESTRICT = "persist:RESTRICT";
+    private static final String CASCADE = "persist:CASCADE";
+    private static final String SET_NULL = "persist:SET_NULL";
+    private static final String NO_ACTION = "persist:NO_ACTION";
+    private static final String RESTRICT_SYNTAX = " RESTRICT";
+    private static final String CASCADE_SYNTAX = " CASCADE";
+    private static final String NO_ACTION_SYNTAX = " NO ACTION";
+    private static final String SET_NULL_SYNTAX = " SET NULL";
+    private static final String SET_DEFAULT_SYNTAX = " SET DEFAULT";
+
+    private SqlScriptGenerationUtils(){}
+
+    public static void generateSqlScript(ArrayList<Entity> entityArray, Path absoluteSourcePath) throws BalException {
+        HashMap<String, List<String>> referenceTables = new HashMap<>();
+        List<String> tableNamesInScript = new ArrayList<>();
+        generateSqlScript(entityArray, referenceTables, absoluteSourcePath, tableNamesInScript);
+    }
+
+    public static void generateSqlScript(ArrayList<Entity> entityArray,
+                                         HashMap<String, List<String>> referenceTables, Path filePath,
+                                         List<String> tableNamesInScript) throws BalException {
+        try {
+            Files.deleteIfExists(Paths.get(String.valueOf(filePath), PersistToolsConstants.FILE_NAME));
+        } catch (IOException e) {
+            throw new BalException("Error while reading the SQL script file (persist_db_push.sql) " +
+                    "generated in the project target directory. ");
+        }
+        for (Entity entity : entityArray) {
+            String tableName = entity.getTableName();
+            String sqlScript = generateDropTableQuery(tableName) + generateTableQuery(tableName, entityArray, entity,
+                    referenceTables);
+            createSqlFile(sqlScript, entity.getTableName(), referenceTables, tableNamesInScript, filePath);
```

Comment:
```suggestion
            updateSqlFile(sqlScript, entity.getTableName(), referenceTables, tableNamesInScript, filePath);
```
Rather wouldnt it better to get create statements in separate function. And combine them in different function